### PR TITLE
[FIX] web_editor, website: fix documents upload

### PR DESCRIPTION
**Pull request by younn-o**
Before this commit, a document uploaded from the Media Dialog was
inserted as a "document" icon link (using the o_image class), without
text.
After that, it was not possible to modify it with the SnippetsMenu link
tools (clicking on it in edit mode had no effect, and the document was
downloaded automatically in normal mode).

Now, the document will be inserted as a regular link, whose content will
be the title of the document. The o_image class is no longer used in
this case.

After, the user can modify the link with the link tools:
- open in the current/new window for documents viewable in the
browser (e.g. pdf documents), downloaded otherwise (pptx, ...),
- change the link style and content.

task-2172311

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/client_actions/website_preview/website_preview.js [hunk 1]
```diff
@@ -255,7 +255,11 @@ export class WebsitePreview extends Component {
      */
     _isTopWindowURL({ host, pathname }) {
         const backendRoutes = ['/web', '/web/session/logout'];
-        return host !== window.location.host || (pathname && (backendRoutes.includes(pathname) || pathname.startsWith('/@/')));
+        return host !== window.location.host
+            || (pathname
+                && (backendRoutes.includes(pathname)
+                    || pathname.startsWith('/@/')
+                    || pathname.startsWith('/web/content')));
```
Comment (qsm-odoo) [inline addons/website/static/src/client_actions/website_preview/website_preview.js:262]: This should be on its own commit (actually probably even on its own PR), it has nothing to do with the rest

File: addons/web_editor/static/src/components/media_dialog/document_selector.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ export class DocumentSelector extends FileSelector {
     static async createElements(selectedMedia, { orm }) {
         return Promise.all(selectedMedia.map(async attachment => {
             const linkEl = document.createElement('a');
-            let href = `/web/content/${attachment.id}?unique=${attachment.checksum}&download=true`;
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/components/media_dialog/document_selector.js:52]: I don't see this commit adding the option to download... was it already there and I don't remember it? :thinking:

File: addons/web_editor/static/src/components/media_dialog/document_selector.js [hunk 3]
```diff
@@ -64,11 +64,16 @@ export class DocumentSelector extends FileSelector {
...
+            linkEl.dataset.fileName = attachment.name;
+            linkEl.textContent = attachment.name;
+            // The document behaviour is by default set to open in a new tab.
+            // This behaviour is configurable with an option in the Link Tools.
+            linkEl.target = '_blank';
             return linkEl;
         }));
     }
 }
-DocumentSelector.mediaSpecificClasses = ['o_image'];
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/components/media_dialog/document_selector.js:71]: Is the `o_image` class still used anywhere else? If not, I guess we can keep it for compatibility but it should be explained next to the related SCSS rules.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/web_editor/static/src/components/media_dialog/document_selector.js (hunk 3 above)
Comment (younn-o) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:71]: Yes, it is still used, at least for knowledge files

File: addons/web_editor/static/src/components/media_dialog/document_selector.js (hunk 2 above)
Comment (younn-o) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:52]: This commit is a minimal JS fix, we'd have to modify the link tools views to add that option, which is developed here in a more complete feature: https://github.com/odoo/odoo/pull/76313

File: addons/website/static/src/client_actions/website_preview/website_preview.js (hunk 1 above)
Comment (younn-o) [reply addons/website/static/src/client_actions/website_preview/website_preview.js:262]: Moved this fix here: https://github.com/odoo/odoo/pull/102225

File: addons/web_editor/static/src/components/media_dialog/document_selector.js (hunk 3 above)
Comment (qsm-odoo) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:71]: Indeed, then ok :+1:

File: addons/web_editor/static/src/components/media_dialog/document_selector.js (hunk 2 above)
Comment (qsm-odoo) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:52]: @younn-o You marked this PR ready for review 1 hour ago but did not reply nor change here? :thinking:

Comment (younn-o) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:52]: Oups, did not receive the notification on that one... 
image
image

For now, the behaviour is the following:
- A document viewable in the browser will be opened in the browser (in a new tab or not)
=> The user can still download it from the browser window  
- If it is not viewable in the browser, it will be downloaded automatically
When discussing with @pierre-pde, it was agreed that this behaviour was making sense.

But it is also possible to keep the `download=true`, say that all documents added from the media dialog are downloaded by default, and hide the "open in a new window" option from the link tools in that case.

@qsm-odoo @pierre-pde What do you think about it ?

Comment (qsm-odoo) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:52]: Ok, that's fine. I'll test and merge :+1:

Comment (younn-o) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:52]: > After testing on Firefox, there seems to be a problem with the LinkPopover. I'll mark the PR as ready for review once solved
@qsm-odoo Unrelated to the PR, the problem is there in 16.0. I have created a task for that bug

File: addons/web_editor/static/src/components/media_dialog/document_selector.js (hunk 3 above)
Comment (qsm-odoo) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:71]: @younn-o If you edit an old o_image (which actually is still used as you explained), it's problematic:
- if you change to another document, you keep the icon but text is added
- if you change to an image, the image dimensions are forced to the o_image one

Can't we improve that? I am not sure I am confident merging this change in 16.0, the first thing I tried is not right :/ Also, is not there a way to avoid the enterprise PR and the change in knowledge?

Comment (younn-o) [reply addons/web_editor/static/src/components/media_dialog/document_selector.js:71]: The o_image class is now removed by default to tackle documents uploaded before this fix.
Unfortunately, with this change in the MediaDialog api, the knowledge override must be adapted...
I rebased https://github.com/odoo/odoo/pull/76313 - which targets master - on top of this fix. If the change in the Media Dialog API is too risky for stable, this fix can always be included in the mentioned PR
